### ISSUE TITLE
Is this project alive ?

### ISSUE THREAD

**Issue by notzoom** [pypyjs/pypyjs#213]

**Comment by jedie** [comment]
Yes, and no ;)

The biggest Problem with pypyjs is the big download size. e.g.: https://github.com/pypyjs/pypyjs/issues/196

WebAssembly is IMHO most interesting for the future, see: #145

**Comment by rfk** [comment]
As several people have noted, and as I'm finally ready to admit to myself, I'm basically not making any time for my personal open-source projects any more:
 
  https://rfk.id.au/blog/entry/archiving-open-source-projects/

For things hosted on my personal github, I'm moving many of them to 'archive' mode in order to make this more clear.

@jedie @crantila and anyone else with a vested interest in this repository - what's the right thing to do here?  Sad as it is, I don't have any reason to believe I'll return to PyPy.js development in the foreseeable future.

I'm tempted to archive this project as well just to help set proper expectations for users, but I'm open to other suggestions.

**Comment by crantila** [comment]
I'm okay with archiving the project. My organization doesn't currently have the resources to use PyPy.js, so we've unfortunately (and hopefully temporarily) returned to a client-server architecture).

Archiving the project would be reversible, if someone should decide they're able to take maintainership in the future?

**Comment by rfk** [comment]
> Archiving the project would be reversible, if someone should decide they're able
> to take maintainership in the future?

Yep, I've double-checked this to make sure.

**Comment by Serkan-devel** [comment]
Will this also affect a potential wasm implementation of it?

**Comment by rfk** [comment]
> Will this also affect a potential wasm implementation of it?

Yes, sadly, it's clear that I'm just not going to make time to work on the wasm implementation any further :-(

**Comment by albertjan** [comment]
no apologies @rfk pypyjs is awesome just 5 years too early 😄any one that has done open source knows what a huge time investment it is. I'm very certain that this project or something similar will popup somewhere sometime again.

**Comment by gabrielfreire** [comment]
hi @albertjan take a look at Pyodide, they are using WebAssembly and is very alive
https://github.com/iodide-project/pyodide
cheers

**Comment by amirouche** [comment]
todomvc demo with pypijs https://github.com/amirouche/pypyjs-todomvc all running in the browser.

Anyone knows how to get websocket working python side?

**Comment by DonaldTsang** [comment]
Question: is this useful? https://github.com/QQuick/Transcrypt

**Comment by amirouche** [comment]
> is Transcrypt useful?

Yes.

**Comment by jedie** [comment]
https://github.com/pmp-p/wapy seems to be interesting: wasm micropython port
See also: https://github.com/pmp-p/micropython-ports-wasm

**Comment by redradist** [comment]
> 
> 
> > Will this also affect a potential wasm implementation of it?
> 
> Yes, sadly, it's clear that I'm just not going to make time to work on the wasm implementation any further :-(

@rfk  It is sad to hear ...

Okay, could you help maybe and explain little bit how you implemented custom JIT backend for PyPy ?

**Comment by redradist** [comment]
@rfk Seems like it is possible to implement PyPy,js with wasm JIT emiter by providing JavaScript functions that will emit WebAssembly code and then PyPy.wasm will call this functions through JavaScript

**Comment by Emmanuel12345678** [comment]
Can I run tensorflow in PyPy or pyodide?

**Comment by jedie** [comment]
> The biggest Problem with pypyjs is the large download size. e.g.: #196
> ...
> https://github.com/pmp-p/wapy seems to be interesting: wasm micropython port
> See also: https://github.com/pmp-p/micropython-ports-wasm

Seems that there is a new project that tried to fix this problem by using MicroPython: https://github.com/pyscript/MicroPyScript
It's around 300k to download.

A DEMO is here: https://pyscript.net/tech-preview/micropython/index.html

See also: https://talkpython.fm/episodes/show/391/pyscript-powered-by-micropython
